AtelierTomato.Markov/AtelierTomato.Markov.Data/Model/OIDPattern.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
+namespace AtelierTomato.Markov.Data.Model
+{
+	public class OIDPattern
+	{
+		public static Regex Generate(IEnumerable<string> fields)

```

Comment:
If you need indexes, take a type that has indexes (i think that is IList or ICollection), that way you don't need to use ElementAt which is a method that should ideally never be used